Orange/widgets/data/tests/test_owfile.py

```diff
@@ -195,3 +195,21 @@ def test_check_datetime_disabled(self):
             for i in range(4):
                 vartype_delegate.setEditorData(combo, idx(i))
                 self.assertEqual(combo.count(), counts[i])
+
+    def test_open_bad_pickle(self):
```

Comment:
This is not the proper test for this code (and not the proper place for the test function itself). You modified `Orange.data.io.PickleReader`, so you should test  `Orange.data.io.PickleReader` and not a widget. You should always aim to test as little code as possible. The test should not rely on the way in which a widget that uses the reader handles exceptions.

Second, simplify this test. You don't need temporary files. Instead of writing a temporary pickle file, just patch `open` and `pickle.load` to return `None`. Then call `self.assertRaises(TypeError, PickleReader.read, "foo")`. You'll do everything in three simple lines.